lib/plug/static.ex

```diff
@@ -182,6 +186,100 @@ defmodule Plug.Static do
     h in only or match?({0, _}, prefix != [] and :binary.match(h, prefix))
   end
 
+  defp serve_range(conn, path, segments, [], %{gzip?: gzip?, brotli?: brotli?} = options) do
+    encoding = file_encoding(conn, path, gzip?, brotli?)
+    serve_static(encoding, segments, options)
+  end
+  defp serve_range(conn, path, segments, [range], options) do
+    encoding = file_encoding(conn, path, false, false)
+    serve_range(encoding, range, segments, options)
+  end
+  defp serve_range(_conn, _path, _segments, _range, _options) do
+    raise InvalidRangeError
```

Comment:
Is raising the reasonable thing to do here? You said we don't support all syntaxes, so maybe we should rather send the whole file instead of crashing.